src/Prometheus/Summary.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Prometheus;
+
+use InvalidArgumentException;
+use Prometheus\Storage\Adapter;
+
+class Summary extends Collector
+{
+    const TYPE = 'summary';
+
+    /**
+     * @var array|null
+     */
+    private $quantiles;
+
+    /**
+     * @param Adapter $adapter
+     * @param string $namespace
+     * @param string $name
+     * @param string $help
+     * @param array $labels
+     * @param array $quantiles
+     * @throws InvalidArgumentException
+     */
+    public function __construct(Adapter $adapter, string $namespace, string $name, string $help, array $labels = [], ?array $quantiles = null)
+    {
+        parent::__construct($adapter, $namespace, $name, $help, $labels);
+
+        if (null === $quantiles) {
+            $quantiles = self::getDefaultQuantiles();
+        }
+
+        if (0 === count($quantiles)) {
```

Comment:
We need to flip these yoda statements to be consistent with the existing code base